inference-engine/tests/ie_test_utils/common_test_utils/ngraph_test_utils.cpp

```diff
@@ -83,6 +85,511 @@ std::string name(const Node& n) {
+    ON_ADAPTER(uint16_t)
+    ON_ADAPTER(uint32_t)
+    ON_ADAPTER(uint64_t)
+    ON_ADAPTER(float)
+    ON_ADAPTER(double)
+    ON_ADAPTER(std::vector<int8_t>)
+    ON_ADAPTER(std::vector<int16_t>)
+    ON_ADAPTER(std::vector<int32_t>)
+    ON_ADAPTER(std::vector<int64_t>)
+    ON_ADAPTER(std::vector<uint8_t>)
+    ON_ADAPTER(std::vector<uint16_t>)
+    ON_ADAPTER(std::vector<uint32_t>)
+    ON_ADAPTER(std::vector<uint64_t>)
+    ON_ADAPTER(std::vector<float>)
+    ON_ADAPTER(std::vector<double>)
+    ON_ADAPTER(std::vector<std::string>)
+
+#undef ON_ADAPTER
+
+    void on_adapter(
+        const std::string&, ngraph::ValueAccessor<std::shared_ptr<ngraph::Function>>&) override {
+        // handled by `compare_functions` drop it here
+    }
+
+    bool all_attr_was_compared() const {
+        return m_visited_attributes.size() == m_attr_ref.attributes_number();
+    }
+
+    size_t compared_attr_number() const {
+        return m_visited_attributes.size();
+    }
+
+    const CmpResult& cmp_result() const {
+        return m_cmp_result;
+    }
+
+private:
+    bool should_return() const {
+        return m_fast_exit && m_cmp_result.has_error();
+    }
+    template <typename AttrValue>
+    void verify(const std::string& name, const AttrValue& attr_value) {
+        if (should_return()) {
+            return;
+        }
+        m_visited_attributes.insert(name);
+        const auto ref_value = m_attr_ref.get<AttrValue>(name);
+        if (!ref_value) {
+            m_cmp_result += "missing attribute name: " + name;
+            return;
+        }
+
+        if (!equal::Equal<AttrValue>::equal_value(*ref_value, attr_value)) {
+            m_cmp_result += "mismatch in value: " + name;
+            return;
+        }
+    }
+
+    const ReadAndStoreAttributes& m_attr_ref;
+    CmpResult m_cmp_result;
+    std::set<AttrName> m_visited_attributes;
+    bool m_fast_exit{true};
+};
+
+}  // namespace attr_comparison
+
+class CompareNodesAttributes {
+public:
+    CompareNodesAttributes() : m_compare_attr(m_store_attr) {}
+
+    attr_comparison::ReadAndStoreAttributes& get_ref_reder() {
+        return m_store_attr;
+    }
+
+    attr_comparison::ReadAndCompareAttributes& get_cmp_reader() {
+        return m_compare_attr;
+    }
+
+    bool equal() const {
+        return m_compare_attr.all_attr_was_compared() && !m_compare_attr.cmp_result().has_error();
+    }
+
+    friend std::string to_string(const CompareNodesAttributes& c) {
+        const auto& result = c.m_compare_attr.cmp_result();
+        if (result.has_error()) {
+            return result.message();
+        }
+        if (!c.m_compare_attr.all_attr_was_compared()) {
+            return "not all of attr was compared: " +
+                   std::to_string(c.m_compare_attr.compared_attr_number()) + " vs " +
+                   std::to_string(c.m_store_attr.attributes_number());
+        }
+        return "looks good [compared " + std::to_string(c.m_compare_attr.compared_attr_number()) +
```

Comment:
Hmmm, I am not sure that we need to print message if attributes are the same, when we compare nodes or anything else we print nothing.